terraform/azure-ghaf-infra.tf

```diff
@@ -44,3 +82,55 @@ resource "azurerm_network_security_group" "ghafhydra_terraform_nsg" {
   }
 }
 
+# Create SA
+
+resource "azurerm_storage_account" "ghafinfra_tf_storage_account" {
+  name                     = "ghafinfrastorage"
+  location                 = var.resource_group_location
+  resource_group_name      = azurerm_resource_group.ghaf_infra_tf_dev.name
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+# Create vm
+
+resource "azurerm_virtual_machine" "ghafinfra_tf" {
+  name                  = "ghafinfratf"
+  location              = var.resource_group_location
+  resource_group_name   = azurerm_resource_group.ghaf_infra_tf_dev.name
+  availability_set_id   = azurerm_availability_set.ghaf_infra_tf_availability_set.id
+  network_interface_ids = [azurerm_network_interface.ghaf_infra_tf_network_interface.id]
+  vm_size               = "Standard_B8ms"
+
+  storage_image_reference {
+    publisher = "canonical"
+    offer     = "0001-com-ubuntu-server-jammy"
+    sku       = "22_04-lts-gen2"
+    version   = "latest"
+  }
+
+  storage_os_disk {
+    name              = "ghafinfratfdisk1"
+    create_option     = "FromImage"
+    caching           = "ReadWrite"
+    managed_disk_type = "Premium_LRS"
+    disk_size_gb      = 512
+  }
+
+  os_profile {
+    computer_name  = "ghafinfratf"
+    admin_username = "karim"
+
+    custom_data = <<-EOF
+              #!/bin/sh
+              echo 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDAKcuBPeS7BJqdTDsKvXbEmO13NV0A+cjpI4TNORsMbB2lxT9n8A/aqSuj4IpPnYBtQrS21Y5n1gfCvHAGe059iSVERS6tUP+D1goPjkDwqlclfmRWCzosdi5cC+k1g87dNTtDhsxPLxoYEgqyiCJxB1cha+GEn7uIGo+/GC9+hKC/ryOl2he20yFVahvymnPYbQ9+Zvsk6LJlmeuSgVBoiTLKx8OFRI7vBEr3+pHuZgyXYSVhO/F58x1tA3x8VxW075ZVCEudAwmKwtXA8uGuxx64V6sg7CX6i+E+SML/TIpXMH0Uq5IF9/rjkC6KYn4q1kHUij4PxYUFkjYX/BlR5OXTTAg00XdurwoPTlKPLnUNFpetK/270SXOojscnoC8w7s4xu4QqlO093UR3oNcMjAKB3tTWbhlKibTheSbUcqHhZBsULuiS8xlLCnKQy1YnhhOMtezFHzjzM14ptmb2MFnw4j5zbBsHw0TUtQzAAQqio+TpaRQnTk7EDoxAY8=' >> /home/karim/.ssh/authorized_keys
+              sed -i 's/PasswordAuthentication yes/PasswordAuthentication no/' /etc/ssh/sshd_config
+              systemctl restart sshd
+              EOF
```

Comment:
This will probably not work.
If the command is executed as root, `/home/karim/.ssh/authorized_keys` will be owned by root which is obviously not correct.

If the command is not executed as root, the sed command to edit the /etc/ssh/sshd_config would fail.

Reply:
Also, the sed replacement is a bit fragile. For instance, what if there are many spaces or tabs before 'yes' ?

Reply:
If there is no better way of doing this in terraform,
perhaps we could completely re-write `/etc/ssh/sshd_config` in the intial VM config?

Once the NixOS config gets installed, the ssh configuration will be overwritten and maintained by the nix flakes configs in this repo.